app/controllers/idv/in_person/usps_locations_controller.rb

```diff
@@ -13,25 +13,25 @@ class UspsLocationsController < ApplicationController
 
       # retrieve the list of nearby IPP Post Office locations with a POST request
       def index
-        usps_response = []
+        response = []
+
         begin
           if IdentityConfig.store.arcgis_search_enabled
             candidate = UspsInPersonProofing::Applicant.new(
               address: search_params['street_address'],
               city: search_params['city'], state: search_params['state'],
               zip_code: search_params['zip_code']
             )
-            usps_response = proofer.request_facilities(candidate)
+            response = proofer.request_facilities(candidate)
           else
-            usps_response = proofer.request_pilot_facilities
+            response = proofer.request_pilot_facilities
           end
-        rescue ActionController::ParameterMissing
-          usps_response = proofer.request_pilot_facilities
-        rescue Faraday::ConnectionFailed => _error
-          nil
+        rescue => err
+          NewRelic::Agent.notice_error(err)
+          response = proofer.request_pilot_facilities
```

Comment:
I see this was mostly the old behavior, but this will retry on all errors? do we want that? I would expect we return an empty response with a warning.

Also if the error is being handled, maybe we just log a warning instead of notifying NewRelic? Since it's not resulting in a 500 to the user?

Reply:
@allthesignals to be clear, LG-8514 is just a short-term fix for now and LG-8281 is intended to be a more robust solution? Basically this PR is just to enable us to get PO search out the door with the pilot locations and LG-8281 will add error handling appropriate for nationwide launch?

If we go this route, of falling back to pilot locations for now, I think we should log whatever the underlying exception was in some fashion. Either NewRelic or as an analytics event. And then when we work LG-8281 we'll have a better sample set of exceptions we've already run into

Reply:
ah yes I see now about the static locations, thanks for clarifying.

I think either analytics events or logger.warn would be clearer than alerting NewRelic, because we associate NewRelic errors with 500s or other user-facing errors, which in this case it's not